pyramid-markdown-it.js

```diff
@@ -0,0 +1,214 @@
+          <div class="pyramid-content">${contentArray[i]}</div>
+        </div>
+      </foreignObject>
+    `;
+  }
+
+  svg += '</svg>';
+  return svg;
+}
+
+/**
+ * Parse options from string format to object
+ */
+function parseOptions(optionsStr) {
+  if (!optionsStr) return {};
+  
+  // Handle space-separated numeric values (width height fontSize)
+  if (/^\d+(\s+\d+){0,2}$/.test(optionsStr)) {
+    const values = optionsStr.split(/\s+/).map(Number);
+    const result = {};
+    if (values.length > 0) result.width = values[0];
+    if (values.length > 1) result.height = values[1];
+    if (values.length > 2) result.fontSize = values[2];
+    return result;
+  }
+  
+  // Handle key=value pairs
+  const options = {};
+  const optionPairs = optionsStr.match(/(\w+)=([^\s]+)/g) || [];
+  
+  optionPairs.forEach(pair => {
+    const [key, value] = pair.split('=');
+    options[key] = /^\d+$/.test(value) ? parseInt(value, 10) : 
+                  /^\d+\.\d+$/.test(value) ? parseFloat(value) :
+                  value === 'true' ? true :
+                  value === 'false' ? false : value;
+  });
+  
+  return options;
+}
+
+/**
+ * Check if a string is a valid hex color code
+ */
+function isColorCode(str) {
+  return str && /^#([0-9A-F]{3}){1,2}$/i.test(str);
+}
+
+/**
+ * Process pyramid diagram content from Mermaid code block
+ */
+function processPyramidDiagram(content) {
+  const lines = content.trim().split('\n');
+  const contentArray = [];
+  const optionsArray = [];
+  let globalOptions = {};
+  
+  // Find the start index (after 'pyramid' line)
+  let startIndex = lines.findIndex(line => line.trim() === 'pyramid') + 1;
+  if (startIndex === 0) startIndex = 1; // Default to 1 if 'pyramid' not found
+  
+  // Check for options line
+  if (startIndex < lines.length && lines[startIndex]?.trim().startsWith('options:')) {
+    globalOptions = parseOptions(lines[startIndex].substring(8).trim());
+    startIndex++;
+  }
+  
+  // Process content lines
+  for (let i = startIndex; i < lines.length; i++) {
+    const line = lines[i].trim();
+    if (!line) continue;
+    
+    // Parse the line content and options
+    const parts = line.split('|').map(part => part.trim());
+    let content = parts[0];
+    let layerOptions = {};
+    
+    // Process options and content from parts
+    if (parts.length > 1) {
+      if (isColorCode(parts[1])) {
+        layerOptions.color = parts[1];
+      } else if (parts[1].includes('=')) {
+        layerOptions = parseOptions(parts[1]);
+      }
+    }
+    
+    if (parts.length > 2) {
+      if (isColorCode(parts[2])) {
+        layerOptions.color = parts[2];
+      } else {
+        content = parts[2];
+      }
+    }
+    
+    contentArray.push(content);
+    optionsArray.push(layerOptions);
+  }
+  
+  return createPyramidSVG(contentArray, optionsArray, globalOptions);
+}
+
+// Export the markdown-it plugin
+module.exports = function markdownItPyramid(md) {
```

Comment:
I suggest splitting the files into

pyramid.js (that just exports the pyramid code and renders the diagram)
chevron.js
venn.js

smartar-markdown-it.js (that uses pyramid.js, etc. and exports a markdown-it plugin).

This can be imported by the marp engine.